src/ctapipe/tools/make_irf.py

```diff
@@ -0,0 +1,615 @@
+        "cuts": "IrfTool.cuts_file",
+        "gamma-file": "IrfTool.gamma_file",
+        "proton-file": "IrfTool.proton_file",
+        "electron-file": "IrfTool.electron_file",
+        "output": "IrfTool.output_path",
+        "chunk_size": "IrfTool.chunk_size",
+    }
+
+    flags = {
+        **flag(
+            "do-background",
+            "IrfTool.do_background",
+            "Compute background rate.",
+            "Do not compute background rate.",
+        ),
+        **flag(
+            "do-benchmarks",
+            "IrfTool.do_benchmarks",
+            "Produce IRF related benchmarks.",
+            "Do not produce IRF related benchmarks.",
+        ),
+        **flag(
+            "full-enclosure",
+            "IrfTool.full_enclosure",
+            "Compute a full-enclosure IRF.",
+            "Compute a point-like IRF.",
+        ),
+    }
+
+    classes = (
+        [
+            EventsLoader,
+        ]
+        + classes_with_traits(BackgroundRateMakerBase)
+        + classes_with_traits(EffectiveAreaMakerBase)
+        + classes_with_traits(EnergyMigrationMakerBase)
+        + classes_with_traits(PsfMakerBase)
+        + classes_with_traits(AngularResolutionMakerBase)
+        + classes_with_traits(EnergyBiasResolutionMakerBase)
+        + classes_with_traits(SensitivityMakerBase)
+    )
+
+    def setup(self):
+        self.opt_result = OptimizationResultStore().read(self.cuts_file)
+
+        if not self.full_enclosure and self.opt_result.theta_cuts is None:
+            raise ToolConfigurationError(
+                "Computing a point-like IRF requires an (optimized) theta cut."
+            )
+
+        check_e_bins = partial(
+            check_bins_in_range,
+            valid_range=self.opt_result.valid_energy,
+            raise_error=self.range_check_error,
+        )
+        self.particles = [
+            EventsLoader(
+                parent=self,
+                kind="gammas",
+                file=self.gamma_file,
+                target_spectrum=self.gamma_target_spectrum,
+            ),
+        ]
+        if self.do_background:
+            if self.proton_file:
+                self.particles.append(
+                    EventsLoader(
+                        parent=self,
+                        kind="protons",
+                        file=self.proton_file,
+                        target_spectrum=self.proton_target_spectrum,
+                    )
+                )
+            if self.electron_file:
+                self.particles.append(
+                    EventsLoader(
+                        parent=self,
+                        kind="electrons",
+                        file=self.electron_file,
+                        target_spectrum=self.electron_target_spectrum,
+                    )
+                )
+            if len(self.particles) == 1:
+                raise RuntimeError(
+                    "At least one electron or proton file required when specifying `do_background`."
```

Comment:
I'd require the proton file actually. Only using electrons for background doesn't make sense.